packages/std/src/testing/mock.rs

```diff
@@ -675,6 +684,70 @@ impl BankQuerier {
     }
 }
 
+#[cfg(feature = "stargate")]
+#[derive(Clone, Default)]
+pub struct IbcQuerier {
+    port_id: String,
+    channels: Vec<IbcChannel>,
+}
+
+#[cfg(feature = "stargate")]
+impl IbcQuerier {
+    /// Create a mock querier where:
+    /// - port_id is the port the "contract" is bound to
+    /// - channels are a list of ibc channels
+    pub fn new(port_id: &str, channels: &[IbcChannel]) -> Self {
+        IbcQuerier {
+            port_id: port_id.to_string(),
+            channels: channels.to_vec(),
+        }
+    }
+
+    pub fn query(&self, request: &IbcQuery) -> QuerierResult {
+        let contract_result: ContractResult<Binary> = match request {
+            IbcQuery::Channel {
+                channel_id,
+                port_id,
+            } => {
+                let channel = self
+                    .channels
+                    .iter()
+                    .find(|c| match port_id {
+                        Some(p) => c.endpoint.channel_id.eq(channel_id) && c.endpoint.port_id.eq(p),
+                        None => {
+                            c.endpoint.channel_id.eq(channel_id)
+                                && c.endpoint.port_id == self.port_id
+                        }
+                    })
+                    .cloned();
+                let res = ChannelResponse { channel };
+                to_binary(&res).into()
```

Comment:
I'm worried about it. I know it is a simplification, but as far as I understand `IbcQuery`, should allow me to query for ports that my contract is not bound to which is not covered by this implementation. It will enable testing some cases but completely ignores others, making merging them unhappy.

I would prefer `IbcQuerier` to be more like:

```rust
type IbcChannels = Vec<IbcChannel>;

struct IbcQuerier {
  bound_port_id: Option<String>,
  ports: HashMap<String, IbcChannels>,
}
```

Then I would add some API (possibly some builder if the complexity becomes an issue) to create an `IbcQuerier` with only a single port to which the contract is bound, or a variant where there are some additional ports (and maybe the contract is not bound to any port - I think it is a possible case, however not sure what should be returned then). It would be more complete according to what contract may be interesting to query.

If I query for the port differing from the one my contract is bound to, I will always get an empty response.

Reply:
Thanks for your feedback. `bound_port_id` is clearer naming and probably better.

However your comment on ports isn't true? Under this implementation, you can create the `IbcQuerier` with a port_id (which is the `bound_port_id`) and a list of channels, and those channels can be associated with any port, which allows you to query for arbitrary ports.

I added test cases for this as well (see the `ibc_querier_channels_matching` test case).

Reply:
You are completely right. I somehow missed the channel structure, I approve this PR then, and the name is not this much of a problem. Sorry for that.